plugin-testlib/src/main/kotlin/io/spine/tools/gradle/testing/GradleProject.kt

```diff
@@ -0,0 +1,117 @@
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package io.spine.tools.gradle.testing
+
+import com.google.errorprone.annotations.CanIgnoreReturnValue
+import io.spine.tools.gradle.task.TaskName
+import java.io.File
+import org.gradle.testkit.runner.BuildResult
+import org.gradle.testkit.runner.GradleRunner
+
+/**
+ * Allows to configure a Gradle project for testing needs.
+ *
+ * The project operates in the given directory and allows executing Gradle tasks.
+ */
+public class GradleProject internal constructor(setup: GradleProjectSetup) {
+
+    /** The arguments passed to [runner]. */
+    private val arguments: RunnerArguments
+
+    /** The runner for executing tasks. */
+    private val runner: GradleRunner
+
+    init {
+        arguments = setup.arguments
+        runner = GradleRunner.create()
+            .withProjectDir(setup.projectDir)
+            .withDebug(setup.debug)
+        if (setup.addPluginUnderTestClasspath) {
+            runner.withPluginClasspath()
+        }
+        if (setup.environment != null) {
+            runner.withEnvironment(setup.environment)
+        }
+        writeSources(setup)
+    }
+
+    public companion object {
+
+        /** The ID of a Java Gradle plugin. */
+        public const val javaPlugin: String = "java"
+
+        /**
+         * Starts creation of a new the project.
+         *
+         * @param projectDir
+         *          file system directory under which the project will be created
+         */
+        @JvmStatic
+        public fun setupAt(projectDir: File): GradleProjectSetup {
+            return GradleProjectSetup(projectDir)
+        }
+
+        private fun writeSources(setup: GradleProjectSetup) {
+            val sources = Sources(setup)
+            sources.write()
+        }
+    }
+
+    /**
+     * Expose [GradleRunner] used by this Gradle project for finer tuning.
+     */
+    public fun runner(): GradleRunner {
+        return runner
+    }
+
+    /**
+     * Obtains the directory of this project.
+     */
+    public fun projectDir(): File {
+        return runner().projectDir
```

Comment:
I'd make `runner` and `projectDir` Kotlin properties.

If, however, this was made with the intent of making Java API `get-`-free, then I support this decision. `JvmName` annotations are quite cumbersome.

Reply:
The intent was to make it `get`-free, but assuming these tools are mostly for using in Kotlin-based tests, I think it would be better to have them as properties.